iroh-resolver/src/unixfs_builder.rs

```diff
@@ -361,6 +480,39 @@ pub async fn add_file<S: Store>(path: &Path, rpc: Option<S>) -> Result<Cid> {
     Ok(root.expect("missing root"))
 }
 
+/// Adds a directory.
+/// - storing the content using `rpc.store`
+/// - returns the root Cid
+/// - wraps into a UnixFs directory to preserve the filename
+pub async fn add_dir<S: Store>(path: &Path, rpc: Option<S>, _recursive: bool) -> Result<Cid> {
+    ensure!(path.is_dir(), "provided path was not a directory");
+
+    let mut dir = DirectoryBuilder::new();
+    dir.with_path(PathBuf::from(&path));
+    let dir = dir.build().await?;
```

Comment:
my only... question/concern with this is that we are potentially opening up a huge number of readers for each entry in the dir. We don't consume those readers until we `dir.encode()`

Not sure how much I should be worrying about this.